syntax = "proto3";

package chirp;

message PutRequest {
    bytes key = 1;
    bytes value = 2;
}

message PutReply {
    int32 contain  = 1;
}

message ContainRequest {
    bytes key = 1;
}

message ContainReply {
    int32 contain  = 1;
}

message GetRequest {
    bytes key = 1; 
}

message GetReply {
    bytes value = 1;
}

message DeleteRequest {
    bytes key = 1;
}

message DeleteReply {
    // Empty because success/failure is signaled via GRPC status.
}

service KeyValueStore {
    rpc put (PutRequest) returns (PutReply) {}
    rpc contain (ContainRequest) returns (ContainReply) {}
    rpc get (stream GetRequest) returns (stream GetReply) {}
    rpc deletekey (DeleteRequest) returns (DeleteReply) {}
}

